test/test_preserve_bundler_env.rb

```diff
@@ -39,6 +39,25 @@ def test_usr2_restart_preserves_bundler_environment
     assert_match("Gemfile.bundle_env_preservation_test", new_reply)
   end
 
+  def test_worker_forking_preserves_bundler_config_path
```

Comment:
Locally, if I undo the change introduced to `prune_bundler`,

```sh
git restore --source HEAD^ -- lib
```

This test is still green. In other words, the test does not appear to guard against a regression for the new functionality.

Do you think it is possible to write this test in a way that it fails on `master` in the way you observed?

Reply:
Bah, I was testing with `ruby -Ilib:test test/test_preserve_bundler_env.rb` which does fail on master, but looks like the same thing inside of `bundle exec` doesn't.

I've force-pushed an update that sets BUNDLE_GEMFILE in the test, which should persuade it to fail in either scenario (and then subsequently be fixed by the BUNDLE_APP_CONFIG change).

Reply:
Awesome, thanks!